platform/linux-generic/include/odp_packet_internal.h

```diff
@@ -248,6 +251,76 @@ static inline void packet_init(odp_packet_hdr_t *pkt_hdr, uint32_t len)
 	pkt_hdr->input = ODP_PKTIO_INVALID;
 }
 
+/**
+ * Copy packet metadata
+ *
+ * @param      src_hdr     Source packet header
+ * @param      dst_hdr     Destination packet header
+ * @param      uarea_copy  Copy user area data. If false, user area pointers
+ *                         are swapped between the packet headers (allowed
+ *                         only when packets are from the same pool).
+ *
+ * @retval 0 on success
+ * @retval >0 when packet user area was truncated
+ */
+static inline int _odp_packet_copy_md(odp_packet_hdr_t *src_hdr,
+				      odp_packet_hdr_t *dst_hdr,
+				      odp_bool_t uarea_copy)
+{
+	int ret = 0;
+
+	dst_hdr->input = src_hdr->input;
+	dst_hdr->dst_queue = src_hdr->dst_queue;
+	dst_hdr->cos = src_hdr->cos;
+	dst_hdr->cls_mark = src_hdr->cls_mark;
+	dst_hdr->user_ptr = src_hdr->user_ptr;
+
+	if (src_hdr->p.input_flags.flow_hash)
+		dst_hdr->flow_hash = src_hdr->flow_hash;
+
+	if (src_hdr->p.input_flags.timestamp)
+		dst_hdr->timestamp = src_hdr->timestamp;
+
+	if (src_hdr->p.flags.lso) {
+		dst_hdr->lso_max_payload = src_hdr->lso_max_payload;
+		dst_hdr->lso_profile_idx = src_hdr->lso_profile_idx;
+	}
+
+	if (src_hdr->p.flags.payload_off)
+		dst_hdr->payload_offset = src_hdr->payload_offset;
+
+	dst_hdr->p = src_hdr->p;
+
+	if (uarea_copy) {
+		pool_t *src_pool = src_hdr->event_hdr.pool_ptr;
+		pool_t *dst_pool = dst_hdr->event_hdr.pool_ptr;
+		uint32_t src_uarea_size = src_pool->param_uarea_size;
+		uint32_t dst_uarea_size = dst_pool->param_uarea_size;
+
+		if (dst_hdr->uarea_addr != NULL && src_hdr->uarea_addr != NULL) {
+			memcpy(dst_hdr->uarea_addr, src_hdr->uarea_addr,
+			       dst_uarea_size <= src_uarea_size ? dst_uarea_size :
```

Comment:
There seems to be a `MIN()` macro available in `odp_macros_internal.h`, that could be used?

Reply:
Fixed in V2.